lib/persistence/mongo/Base.js

```diff
@@ -0,0 +1,115 @@
+var Q = require("q");
+var mongo_promise = require('./client.js');
+var crypto = require('crypto');
+var RecordNotFoundError = require('./RecordNotFound')
+
+module.exports = function(options){
+
+  var name = options['name'];
+  if ( !name ) { throw new Error("name is required"); }
+
+  var model = options['model'];
+  if ( !model ) { throw new Error("model is required"); }
+
+  var mongo_model = require('../backbone_sync.js')(model)
+
+  mongo_model = mongo_model.extend({
+
+    pk: function(){
+      return this.get('_id');
+    },
+
+    idAttribute: "_id",
+
+    create: function(attributes, options){
+
+      if(!this.collection_name){
+        throw new Error("need the collection_name attribute on the model!")
+      }
+
+      // 24 character id string, as opposed to using mongo's builtin ObjectID
+      this.set('_id', crypto.randomBytes(12).toString('hex'))
```

Comment:
My sense (and I could be wrong) is that ObjectID's are only partly conventional. Where you would abandon them is when you have slightly custom requirements for ids (e.g. they are not just a key to lookup and join documents).

See e.g. meteor's coice to use strings:
https://groups.google.com/d/msg/meteor-talk/f-ljBdZOwPk/oQYZQxCAKN8J
https://groups.google.com/d/msg/meteor-talk/f-ljBdZOwPk/oQYZQxCAKN8J